### PR TITLE
Fix the function selection panel not reloading when changing SAM templates (Fixes #955)

### LINKED ISSUE #955: Edit Run/Debug configuration panel not working correctly for Function Selection Combo box.
**To reproduce**
1. Create a new Run/Debug configuration opens the edit configuration panel.
1. Select **From template:** and select one function from the combo box to the right
1. Click **SAM CLI** tab and type something to the **Docker Network**
1. An error shows in the below that says `Error: Function logical name must be specified.`

**Expected behavior**
The function combo box should not be reloaded.

**Screenshots**
Screen Shot 2019-05-09 at 2 47 16 PM

**Your Environment**
 - OS: Mac
 - JetBrains' Product: IntelliJ
 - JetBrains' Product Version: 2019.1
 - Toolkit Version: 1.3
 - SAM CLI Version:
 - JVM/Python Version:

**Additional context**

### PR DESCRIPTION
**Pull request by hunterwerlla**
- Fix #955, reload the selection box when the selected template changes
- Convert the panel to Kotlin and fixup some conversion weirdness

Actual change is:
```kt
            ProjectFileBrowseListener(
                project,
                templateFile,
                FileChooserDescriptorFactory.createSingleFileDescriptor(YAMLFileType.YML),
                TextComponentAccessor.TEXT_FIELD_WHOLE_TEXT
            ) {
               setTemplateFile(it.canonicalPath) <-- Adding this callback
            }
```
## Related Issue(s)
#955

## License
I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: jetbrains-core/src/software/aws/toolkits/jetbrains/services/lambda/execution/local/LocalLambdaRunSettingsEditorPanel.kt [hunk 1]
```diff
@@ -0,0 +1,167 @@
...
+        runtimeModel = SortedComboBoxModel(compareBy(Comparator.naturalOrder()) { it: Runtime -> it.toString() })
+        runtime = ComboBox(runtimeModel)
+        environmentVariables = EnvironmentVariablesTextField()
+        timeoutSlider = lambdaTimeout()
+        memorySlider = lambdaMemory()
+    }
+
+    init {
+        lambdaInputPanel.border = IdeBorderFactory.createTitledBorder(message("lambda.input.label"), false, JBUI.emptyInsets())
+        useTemplate.addActionListener { e: ActionEvent? -> updateComponents() }
```
Comment (abrooksv) [inline jetbrains-core/src/software/aws/toolkits/jetbrains/services/lambda/execution/local/LocalLambdaRunSettingsEditorPanel.kt:72]: Can remove e

File: jetbrains-core/src/software/aws/toolkits/jetbrains/services/lambda/execution/local/LocalLambdaRunSettingsEditorPanel.kt [hunk 2]
```diff
@@ -0,0 +1,167 @@
...
+            ProjectFileBrowseListener(
+                project,
+                templateFile,
+                FileChooserDescriptorFactory.createSingleFileDescriptor(YAMLFileType.YML),
+                TextComponentAccessor.TEXT_FIELD_WHOLE_TEXT
+            ) {
+                setTemplateFile(it.canonicalPath)
+            }
+        )
+        runtime.addActionListener { e: ActionEvent? ->
```
Comment (abrooksv) [inline jetbrains-core/src/software/aws/toolkits/jetbrains/services/lambda/execution/local/LocalLambdaRunSettingsEditorPanel.kt:84]: Can remove e

File: jetbrains-core/src/software/aws/toolkits/jetbrains/services/lambda/execution/local/LocalLambdaRunSettingsEditorPanel.kt [hunk 3]
```diff
@@ -0,0 +1,167 @@
...
+        functionModels.selectedItem = function
+        updateComponents()
+    }
+
+    fun setRuntimes(runtimes: List<Runtime>?) {
+        runtimeModel.setAll(runtimes)
+    }
+
+    fun invalidateConfiguration() {
+        SwingUtilities.invokeLater { invalidator.isSelected = !invalidator.isSelected }
```
Comment (abrooksv) [inline jetbrains-core/src/software/aws/toolkits/jetbrains/services/lambda/execution/local/LocalLambdaRunSettingsEditorPanel.kt:165]: why not runInEdit?

**Review by abrooksv — COMMENTED** [review]
What was the actual fix? Hard to see it in the Kotlin conversion

File: jetbrains-core/src/software/aws/toolkits/jetbrains/services/lambda/execution/local/LocalLambdaRunSettingsEditorPanel.kt (hunk 1 above)
Comment (hunterwerlla) [reply jetbrains-core/src/software/aws/toolkits/jetbrains/services/lambda/execution/local/LocalLambdaRunSettingsEditorPanel.kt]: Removed the one below, missed this one. Removed

File: jetbrains-core/src/software/aws/toolkits/jetbrains/services/lambda/execution/local/LocalLambdaRunSettingsEditorPanel.kt (hunk 2 above)
Comment (hunterwerlla) [reply jetbrains-core/src/software/aws/toolkits/jetbrains/services/lambda/execution/local/LocalLambdaRunSettingsEditorPanel.kt]: ditto, removed

File: jetbrains-core/src/software/aws/toolkits/jetbrains/services/lambda/execution/local/LocalLambdaRunSettingsEditorPanel.kt (hunk 3 above)
Comment (hunterwerlla) [reply jetbrains-core/src/software/aws/toolkits/jetbrains/services/lambda/execution/local/LocalLambdaRunSettingsEditorPanel.kt]: This is what was here before, didn't change it. Changed to `runInEdt`

**Review by hunterwerlla — COMMENTED** [review]
Added the actual line changed into the description:
```kt
            ProjectFileBrowseListener(
                project,
                templateFile,
                FileChooserDescriptorFactory.createSingleFileDescriptor(YAMLFileType.YML),
                TextComponentAccessor.TEXT_FIELD_WHOLE_TEXT
            ) {
               setTemplateFile(it.canonicalPath) <-- Adding this callback
            }
```

**Review by abrooksv — APPROVED** [review]

**Review by abrooksv — APPROVED** [review]

**Pull request merged**